algorithms/data_structures/lcp_array.py

```diff
@@ -0,0 +1,86 @@
+import math
+
+"""
+    Suffix Array
+    ------------------
+    In computer science, a suffix array is a sorted array of all suffixes
+    of a string. It is a data structure used, among others, in full text
+    indices, data compression algorithms and  within the field
+    of bioinformatics.
+
+    for more info : http://algs4.cs.princeton.edu/63suffix/
+    Complexity :
+        worst case : O(n log(n))
+"""
+
+
+def suffix_array(t_string):
+    t_length = len(t_string)
+    rank = [0] * t_length
+    s_array = [0] * t_length
+    tuple_array = [0] * t_length
+    iterations = int(math.log(t_length, 2)) + 1
+    size = 1
+
+    for i, s in enumerate(t_string):
+        s_array[i] = ord(s)
+
+    for _ in range(iterations):
+        for i, ele in enumerate(tuple_array):
+            if i + size < t_length:
+                tuple_array[i] = ((s_array[i], s_array[i + size]), i)
+            else:
+                tuple_array[i] = ((s_array[i], -1), i)
+        tuple_array.sort()
+        s_array[tuple_array[0][1]] = 0
+        for i in range(1, len(tuple_array)):
+            cls, idx = tuple_array[i]
+            if cls == tuple_array[i - 1][0]:
+                s_array[idx] = s_array[tuple_array[i - 1][1]]
+            else:
+                s_array[idx] = s_array[tuple_array[i - 1][1]] + 1
+        size *= 2
+
+    for i, p in enumerate(s_array):
+        rank[p] = i
+
+    return s_array, rank
+
+
+"""
+    LCP Array
+    ------------------
+    the longest common prefix array (LCP array) is an auxiliary data
+    structure to the suffix array. It stores the lengths of the longest
+    common prefixes (LCPs) between all pairs of consecutive suffixes
+    in a sorted suffix array.
+
+    I use Kasai's algorithm in implementation :
+    Pseudo Code: http://algs4.cs.princeton.edu/32bst
+
+    Complexity :
+     worst case :O(n)
+
+"""
+
+
+def lcp_array(t_str, s_array, rank):
```

Comment:
Can you add doc strings? Take a look at few of the other modules to get an idea of what should be expected.